Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Internal/DriverReferee.hs

```diff
@@ -61,6 +67,13 @@ linkReferee merchantId apiKey RefereeLinkInfoReq {..} = do
   currency <- maybe (pure INR) SMerchant.getCurrencyByMerchantOpCity driverInfo.merchantOperatingCityId
   unless (driverInfo.enabled) $
     throwError $ InvalidRequest "Driver is not enabled"
+  driverStats <- QDriverStats.findByPrimaryKey driverReferralLinkage.driverId >>= fromMaybeM (PersonNotFound driverReferralLinkage.driverId.getId)
+  QDriverStats.updateTotalReferralCount (driverStats.totalReferralCounts + 1) driverReferralLinkage.driverId
+  merchOpCityId <- maybe (QP.findById driverReferralLinkage.driverId >>= fromMaybeM (PersonNotFound (driverReferralLinkage.driverId.getId)) <&> (.merchantOperatingCityId)) pure driverInfo.merchantOperatingCityId
+  transporterConfig <- SCTC.findByMerchantOpCityId merchOpCityId (Just (DriverId (cast driverReferralLinkage.driverId))) >>= fromMaybeM (TransporterConfigNotFound merchOpCityId.getId)
+  localTime <- getLocalCurrentTime transporterConfig.timeDiffFromUtc
+  dailyStats <- QDailyStats.findByDriverIdAndDate driverReferralLinkage.driverId (utctDay localTime) >>= fromMaybeM (InvalidRequest "Daily Stats Not Found")
+  QDailyStats.updateReferralCount (dailyStats.referralCounts + 1) driverReferralLinkage.driverId (utctDay localTime)
```

Comment:
is `updateReferralCount` is called at multiple locations if yes the put it in lock